interview-questions.md

```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: JSX stands for JavaScript XML or Javascript Syntax Extension and it makes it easier to write react applications. As it is a markdown syntax extension to javascript. Also, it behaves very similarly to HTML which a couple of differences. Uses syntatical sugar for react, which makes pieces of code "sweeter". In addition, two syntax difference between JSX and HTML is that JSX has reserved keywords for class and for unlike HTML. So to define class names and for attributes use camelCasing; className and HTMLFor.
 
-1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
+2. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn is a package manager that allows people to share code with each other. The node Modules are modified when the command yarn is ran within the terminal.
 
-Researched answer:
+Researched answer: Yarn is a package manager that manages all dependancy files. Which leads to to package manager will install dependencies, manage the dependencies, and supply us terminal commands to execute tests. The file that is modified is the package.json file, and installs node modules with their dependencies. 
 
 3. What is a React component?
 
-Your answer:
+Your answer: A react component is a function that is written in javascript that can be passed to JSX land in between the react fragments.
 
-Researched answer:
+Researched answer: A React Component is a specific type of function that can contain JavaScript logic and returns a markup for the browser to render. In addition, they are independent and reusable bits of code and returns to HTML. It can be either state or stateless, which refers to it can be either a logic component (holds state) or display component (stateless).
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: The difference between state values and props is that state values has two parameters and is a react hook. One that holds the data, and one that changes the data. Props on the other hand, refer to the state values and can be called upon with curly braces within any file.
 
-Researched answer:
+Researched answer: A state value is a special variable that stores data that determines how components renders and behaves. By using the React Hook which is a built-in method, the React Hook useState() can define the variable with an inital value and provide methods to change that value. In contrast, props is short for properties, and is the communication between reusable components in react i.e React Hooks. It passes data from one component down to a nested component, and the data is usually comes from a value of a State. In comparison, a state value holds information and a prop calls upon it and passes it down to components tasked to deal with user interaction of that information from state.
```

Comment:
Kind of confusing answer here. You used React hooks as an example for reusable components, which is not a good example, as hooks can be used in components, but are not components themselves. Here is a general breakdown of state vs props:
State values are set and re-set by using react hooks, and are mutable, which means they can be modified. On the other hand, props are passed from a component to a nested component in order to allow the nested component access to data existing in the original component. Props are also read-only and immutable in the nested component.